sdk/core/azure-core-amqp/inc/azure/core/amqp/network/transport.hpp

```diff
@@ -16,60 +17,119 @@ namespace Azure { namespace Core { namespace Amqp { namespace Network { namespac
 }}}}} // namespace Azure::Core::Amqp::Network::_detail
 
 namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
-  enum class TransportState
-  {
-    Closed,
-    Closing,
-    Open,
-    Opening,
-    Unknown
-  };
 
-  enum class TransportOpenResult
+  /** @brief Status of a transport open operation.
+   *
+   *
+   */
+  enum class TransportOpenStatus
   {
+    /// Transport is in invalid state.
     Invalid,
+    /// Transport open successful.
     Ok,
+    /// Transport open failed with an error.
     Error,
+    /// Transport open was cancelled.
     Cancelled
   };
-  enum class TransportSendResult
+
+  /** @brief Status of a transport send operation.
+   *
+   */
+  enum class TransportSendStatus
   {
+    /// Send status Invalid - this should never be returned.
     Invalid,
+
+    /// Send status unknown
     Unknown,
+
+    /// Send successful
     Ok,
+
+    /// Send completed with an error.
     Error,
+
+    /// Send was cancelled.
     Cancelled
   };
 
   class Transport;
 
+  /** Transport operational events.
+   *
+   * Normally a client will not have to register for these events, but they are available for
+   * clients if necessary.
+   */
   struct TransportEvents
```

Comment:
virtual destructor, optionally turn into `class`

Reply:
Same question as before: Why should a function which only has pure virtual methods be (a) a class and (b) have a destructor?

Reply:
class vs struct: a patern across the SDK - structs are typically value holders, mostly with public fields, maybe there are some with simple getters/setters, while classes have "methods", virtual functions, are designed to be inheritable etc.

Virtual destructors: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-dtor

Reply:
Please file an issue against the C++ guidelines to reflect that pattern. 

Traditionally, for C++ developers, a struct and class are the exact same, the only difference between the two is visibility of fields. If the C++ SDK has adopted a pattern that indicates that structs should never contain methods, that is not an expected behavior for C++  developers and we need to clearly express that difference in our guidelines.